### DISCUSSION TITLE
ShopListing.createDraftListing

### DISCUSSION THREAD

**Discussion by andrewhodel** [etsy/open-api#1088] [Ideas]
1. dpi is a missing field
2. orientation is a missing field (this can be determined after the API request is responded to)
3. the width and height fields are not for each digital file available, you should document that it is an approximation, minimum or maximum or allow that value to be set for each digital file

Thanks.

**Comment by emergingdzns** [comment]
@andrewhodel I don't follow any of this. Can provide some context on what you are wanting here?

**Comment by andrewhodel** [comment]
## dpi field

The width and height fields are only in units that are not pixels.  The value of those fields depends on the DPI that is chosen at print time.

If you don't only show pixel values in the width and height fields you need to have a dpi field to inform the customer.

If the item lister does not specify the DPI and says the image is 30 inches by 30 inches because there is no dpi field, he may have calculated the size in inches using a DPI of 75 while the customer is expecting a DPI of 1200.  Without pixels as the measuring unit or a dpi field, the customer may get a low quality print.

## orientation field

The orientation field is available in the browser UI but not in the API documentation on each listing.

## only one width and height field per listing

There is a width and height field for each listed item, but each listed item can have multiple widths and heights because there are multiple digital files (maximum 5) for each listed item.

**Comment by andrewhodel** [comment]
The ideal solution to all three is to remove all those fields from the input side (regardless of API or UI being the conduit) and calculate the values when the digital files are uploaded then providing those values as output.

**Reply by emergingdzns** [reply]
Ah I see what you are saying. I'm really not sure of the consequences of doing this. For one thing, not all digital download files are images so being able to pull the dpi and dimensions would involve a lot on Etsy's side. I doubt they will go for it. But if you want to request it you would need to email developer@etsy.com and see if they will put in on their radar.

**Reply by andrewhodel** [reply]
Imagemagick is written in C, portable and reads every file type that Etsy supports.

Vector files have a maximum printing dpi that can be understood most easily by building a map tile generator that uses vector SHP files and outputs raster tiles.  Eventually they become poor in quality because curves shown are not valid at higher resolutions without drawing giant tiles and then cutting them into parts.

That means that if you have a problem with vector files at a high resolution, that you shouldn't be parsing a single tile at a time.

That is simply not working on that project beyond drawing single tiles.

What you have to understand is that also flips and makes the SVG file size requirements huge at far zooms.  This is because the coastline data of a whole country must include every inlet and you are only viewing the country as a whole.  What happens is that it takes forever to load one file when they are right because everything has to be rendered.

Again, this is just measuring the maximum length on 2 axes of every curve and line within the vector file.

It's not that difficult, it's nothing more than work through time.

**Comment by andrewhodel** [comment]
Regardless of you leading that program but not explaining the details because there are 2 considerations as solutions.

If you will not implement the difficult and full solution, you should at least provide the missing fields as I requested in the first post to provide more certainty to the customer and create less customer support requirements from the lister.

**Comment by andrewhodel** [comment]
I sent the URL to this thread to developer@etsy.com

**Reply by emergingdzns** [reply]
To be clear, I don't work for Etsy. I can't do anything on their end. When I first commented I thought you were asking for help from the community but I didn't understand what you meant. Sorry for the confusion.

**Comment by andrewhodel** [comment]
I was only trying to prevent customers emailing me because they didn't understand DPI and there weren't fields validating it on Etsy, after they got a print they didn't expect.

If I made everything 1200dpi the new printers would be 4800 dpi.
